src/internal_events/aws_cloudwatch_logs_subscription_parser.rs

```diff
@@ -21,13 +22,79 @@ impl InternalEvent for AwsCloudwatchLogsSubscriptionParserError {
     fn emit_metrics(&self) {
         counter!(
             "component_errors_total", 1,
-            "error" => self.error.to_string(),
+            "error_code" => "parsing_cloudwatch_logs",
             "error_type" => error_type::PARSER_FAILED,
             "stage" => error_stage::PROCESSING,
         );
         // deprecated
-        counter!("processing_errors_total", 1,
+        counter!(
+            "processing_errors_total", 1,
             "error_type" => "failed_parse",
         );
     }
 }
+
+#[derive(Debug)]
+pub struct AwsCloudwatchLogsMessageSizeError {
+    pub size: usize,
+    pub max_size: usize,
+}
+
+impl InternalEvent for AwsCloudwatchLogsMessageSizeError {
+    fn emit_logs(&self) {
+        error!(
+            message = %format!("Encoded event is too long: {} > {}", self.size, self.max_size),
+            error_code = "message_too_long",
```

Comment:
I could see us pulling up some of these more general encoding errors into a shared spot.